test/common/TestBedChild.cpp

```diff
@@ -177,6 +178,29 @@ namespace RcclUnitTesting
 	  break;
 	}
       }
+      else if (this->useBlocking == false)
+      { 
+        // When non-blocking communicator is desired call ncclCommInitRankConfig with appropriate flag
+        ncclResult_t ncclAsyncErrInit;
+        ncclConfig_t config = NCCL_CONFIG_INITIALIZER;
+        config.blocking = 0;
+        if (ncclCommInitRankConfig(&this->comms[localRank], this->totalRanks, id, globalRank, &config) != ncclSuccess)
+        { 
+          ERROR("Rank %d on child %d unable to call ncclCommInitRankConfig\n", globalRank, this->childId);
+          status = TEST_FAIL;
+          break;
+        } 
+        do 
+        {
+          ncclCommGetAsyncError(this->comms[localRank], &ncclAsyncErrInit);
+        } while(ncclAsyncErrInit != ncclSuccess);
+        if (ncclAsyncErrInit != ncclSuccess)
+        {
+          ERROR("Rank %d on child %d fails NCCL call %s with code %d\n", globalRank, this->childId, "ncclCommGetAsyncErrorInit", ncclAsyncErrInit);
+          status = TEST_FAIL;
+          break; 
```

Comment:
To actually "test" out this feature, I think we can move this outside of the local rank loop (and only tested if nonblocking init requested).  Additionally, we need to have some sort of counter to prevent infinite loop, if for some reason, ncclCommGetAsyncError never reaches ncclSuccess.